src/metadata.rs

```diff
@@ -62,7 +62,7 @@ pub fn fetch_metadata(provider: &str) -> errors::Result<Box<dyn providers::Metad
         "ibmcloud" => box_result!(IBMGen2Provider::try_new()?),
         // IBM Cloud - Classic infrastructure.
         "ibmcloud-classic" => box_result!(IBMClassicProvider::try_new()?),
-        "openstack-metadata" => box_result!(OpenstackProvider::try_new()?),
+        "openstack" | "openstack-metadata" => openstack::try_config_drive_else_network(),
```

Comment:
I'd suggest to keep the two providers split and directly wire `openstack-metadata` to `OpenstackProviderNetwork`.